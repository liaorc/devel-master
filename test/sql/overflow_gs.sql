--#
--#       Gpu Scan OverFlow TestCases. 
--#

set enable_seqscan to off;
set enable_bitmapscan to off;
set enable_indexscan to off;
set random_page_cost=1000000;   --# force off index_scan.
set pg_strom.enable_gpuhashjoin to off;
set pg_strom.enable_gpupreagg to off;
set pg_strom.enable_gpusort to off;
set client_min_messages to warning;

-- overflow
select  (smlint_x+1)::smallint    from strom_overflow_test order by id limit 100;
select  (integer_x+1)::integer    from strom_overflow_test order by id limit 100;
select  (bigint_x+1)::bigint    from strom_overflow_test order by id limit 100;
select  (real_x^2)::real    from strom_overflow_test order by id limit 100;
select  (float_x^2)::float    from strom_overflow_test order by id limit 100;
select  (nume_x^10000)::numeric    from strom_overflow_test order by id limit 100;
select  (smlsrl_x^2)::smallint    from strom_overflow_test order by id limit 100;
select  (serial_x^2)::integer    from strom_overflow_test order by id limit 100;
select  (bigsrl_x^2)::bigint    from strom_overflow_test order by id limit 100;

