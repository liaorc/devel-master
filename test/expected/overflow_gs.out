--#
--#       Gpu Scan OverFlow TestCases. 
--#
set enable_seqscan to off;
set enable_bitmapscan to off;
set enable_indexscan to off;
set random_page_cost=1000000;   --# force off index_scan.
set pg_strom.enable_gpuhashjoin to off;
set pg_strom.enable_gpupreagg to off;
set pg_strom.enable_gpusort to off;
set client_min_messages to warning;
-- overflow
select  (smlint_x+1)::smallint    from strom_overflow_test order by id limit 100;
ERROR:  smallint out of range
select  (integer_x+1)::integer    from strom_overflow_test order by id limit 100;
ERROR:  integer out of range
select  (bigint_x+1)::bigint    from strom_overflow_test order by id limit 100;
ERROR:  bigint out of range
select  (real_x^2)::real    from strom_overflow_test order by id limit 100;
ERROR:  value out of range: overflow
select  (float_x^2)::float    from strom_overflow_test order by id limit 100;
ERROR:  value out of range: overflow
select  (nume_x^10000)::numeric    from strom_overflow_test order by id limit 100;
ERROR:  value overflows numeric format
select  (smlsrl_x^2)::smallint    from strom_overflow_test order by id limit 100;
ERROR:  smallint out of range
select  (serial_x^2)::integer    from strom_overflow_test order by id limit 100;
ERROR:  integer out of range
select  (bigsrl_x^2)::bigint    from strom_overflow_test order by id limit 100;
ERROR:  bigint out of range
